systemtest/src/test/java/io/strimzi/systemtest/olm/OlmAllNamespaceST.java

```diff
@@ -21,52 +27,134 @@
 
     public static final String NAMESPACE = "olm-namespace";
 
     @Test
     @Order(1)
+    @TestDoc(
+        description = @Desc("Verifies the deployment of a Kafka cluster using the OLM example when the operator watches all namespaces."),
+        steps = {
+            @Step(value = "Deploy Kafka cluster using the OLM example.", expected = "Kafka cluster is deployed and becomes ready."),
+            @Step(value = "Ensure operator (watching all namespaces) manages the Kafka cluster in the designated test namespace.", expected = "Kafka cluster is operational and managed by the operator.")
+        },
+        labels = {
+            @Label(TestDocsLabels.OLM),
+            @Label(TestDocsLabels.KAFKA)
+        }
+    )
     void testDeployExampleKafka() {
         doTestDeployExampleKafka();
     }
 
     @Test
     @Order(2)
+    @TestDoc(
+        description = @Desc("Verifies the deployment of a KafkaUser using the OLM example when the operator watches all namespaces."),
+        steps = {
+            @Step(value = "Deploy KafkaUser using the OLM example.", expected = "KafkaUser is deployed and becomes ready."),
+            @Step(value = "Ensure operator (watching all namespaces) manages the KafkaUser in the designated test namespace.", expected = "KafkaUser is operational and managed by the operator.")
+        },
+        labels = {
+            @Label(TestDocsLabels.OLM)
+        }
+    )
     void testDeployExampleKafkaUser() {
         doTestDeployExampleKafkaUser();
     }
 
     @Test
     @Order(3)
+    @TestDoc(
+        description = @Desc("Verifies the deployment of a KafkaTopic using the OLM example when the operator watches all namespaces."),
+        steps = {
+            @Step(value = "Deploy KafkaTopic using the OLM example.", expected = "KafkaTopic is deployed and becomes ready."),
+            @Step(value = "Ensure operator (watching all namespaces) manages the KafkaTopic in the designated test namespace.", expected = "KafkaTopic is operational and managed by the operator.")
+        },
+        labels = {
+            @Label(TestDocsLabels.OLM),
+            @Label(TestDocsLabels.KAFKA)
+        }
+    )
     void testDeployExampleKafkaTopic() {
         doTestDeployExampleKafkaTopic();
     }
 
     @Test
     @Order(4)
     @Tag(CONNECT)
+    @TestDoc(
+        description = @Desc("Verifies the deployment of a KafkaConnect cluster using the OLM example when the operator watches all namespaces."),
+        steps = {
+            @Step(value = "Deploy KafkaConnect cluster using the OLM example.", expected = "KafkaConnect cluster is deployed and becomes ready."),
+            @Step(value = "Ensure operator (watching all namespaces) manages the KafkaConnect cluster in the designated test namespace.", expected = "KafkaConnect cluster is operational and managed by the operator.")
+        },
+        labels = {
+            @Label(TestDocsLabels.OLM),
+            @Label(TestDocsLabels.CONNECT)
```

Comment:
Again here, I would not add the label for each of the resource, as I think it doesn't test any functionality of the particular operand, rather than functionality of the OLM and the operator that is able to deploy and manage particular resource. WDYT @Frawless ?